tests/nosetests/pyanaconda_tests/modules/payloads/payload/rpm_ostree_tasks_test.py

```diff
@@ -325,3 +328,64 @@ def run_test(self):
                 for gpg_verify in (True, False):
                     self._execute_run_once(use_sysroot, gpg_verify, verify_ssl)
+
+
+class ConfigureBootloaderTaskTestCase(unittest.TestCase):
+    @patch("pyanaconda.modules.payloads.payload.rpm_ostree.installation.safe_exec_with_redirect")
+    @patch("pyanaconda.modules.payloads.payload.rpm_ostree.installation.os.rename")
+    @patch("pyanaconda.modules.payloads.payload.rpm_ostree.installation.os.symlink")
+    @patch("pyanaconda.modules.payloads.payload.rpm_ostree.installation.STORAGE")
+    @patch("pyanaconda.modules.payloads.payload.rpm_ostree.installation.DeviceData")
+    def _execute_run_once(self, is_dirinstall, is_btrfs_subvolume,
+                          devdata_mock, storage_mock, symlink_mock, rename_mock, exec_mock):
+        proxy_mock = storage_mock.get_proxy()
+        proxy_mock.GetArguments.return_value = ["BOOTLOADER-ARGS"]
+        proxy_mock.GetFstabSpec.return_value = "FSTAB-SPEC"
+        proxy_mock.GetRootDevice.return_value = "device-name"
+        if is_btrfs_subvolume:
+            devdata_mock.from_structure.return_value.type = "btrfs subvolume"
+        else:
+            devdata_mock.from_structure.return_value.type = "something-non-btrfs-subvolume-ish"
+
+        with tempfile.TemporaryDirectory() as sysroot:
+            os.makedirs(sysroot + "/boot/grub2")
+            os.mknod(sysroot + "/boot/grub2/grub.cfg")
+
+            task = ConfigureBootloader(sysroot, is_dirinstall)
+            task.run()
+
+            rename_mock.assert_called_once_with(
+                sysroot + "/boot/grub2/grub.cfg",
+                sysroot + "/boot/loader/grub.cfg"
+            )
+            symlink_mock.assert_called_once_with(
+                "../loader/grub.cfg",
+                sysroot + "/boot/grub2/grub.cfg"
+            )
+            if is_dirinstall:
+                exec_mock.assert_not_called()
+            else:
+                if is_btrfs_subvolume:
+                    exec_mock.assert_called_once_with(
+                        "ostree",
+                        ["admin", "instutil", "set-kargs", "BOOTLOADER-ARGS", "root=FSTAB-SPEC",
+                         "rootflags=subvol=device-name"],
+                        root=sysroot
+                    )
+                else:
+                    exec_mock.assert_called_once_with(
+                        "ostree",
+                        ["admin", "instutil", "set-kargs", "BOOTLOADER-ARGS", "root=FSTAB-SPEC"],
+                        root=sysroot
+                    )
+
+    def run_test(self):
+        """Test OSTree bootloader config task"""
+        # pylint: disable=no-value-for-parameter
+        # non-BTRFS
+        self._execute_run_once(False, False)
+        # BTRFS
+        self._execute_run_once(False, True)
+        # dirinstall
+        self._execute_run_once(True, False)
+        # last combination is impossible, dirinstall and BTRFS are mutually exclusive
```

Comment:
In general this is a anti-pattern. Giving boolean values to a method where all the logic is decided in general means that the logic should be split to a multiple parts and used that instead.

I think that these should be 3 tests:
``non-btrfs-test``
``btrfs-test``
``dir-install-test``

That would be much easier for debugging etc.

Reply:
Done!